### PR TITLE
Step 18.75, let the user choose a TTS

### PR DESCRIPTION
**Pull request by mohammad-suliman**

### CODE REVIEW

File: ts/view/audio.ts [hunk 1]
```diff
@@ -143,5 +143,16 @@ function speakSelectedCell() {
...
+  utterance.voice = selectedTtsVoice;
   synth.speak(utterance);
 }
+
+// In Chrome, we need to wait for the "voiceschanged" event to be fired before we can get the list of all voices. See
+//https://developer.mozilla.org/en-US/docs/Web/API/Web_Speech_API/Using_the_Web_Speech_API
+// for more details 
+let synth = window.speechSynthesis;
+if (synth.onvoiceschanged !== undefined) {
+  synth.onvoiceschanged = speakSelectedCell;
```
Comment (oferb) [inline ts/view/audio.ts]: So when `onvoiceschanged()` is called, we speak the selected cell?
That's kind of like on page load, right?
So when the page loads, we speak the selected cell?

I thought we speak only following user interaction.

File: ts/app.ts [hunk 2]
```diff
@@ -1,13 +1,21 @@
-function setFocusToInputField() {
+jQuery(window).on('load', function () {
```
Comment (oferb) [inline ts/app.ts]: Why did you change this?
Looks like now this code will get called on every page that loads `app.js`. Is that the intention?

File: ts/view/audio.ts (hunk 1 above)
Comment (mohammad-suliman) [reply ts/view/audio.ts]: First, I have to admit that this mechanism is not so clear to me...
From what I have understood, the onvoiceschanged event is fired when the list of voices of the speech API changes, see this for a bit more details:
https://developer.mozilla.org/en-US/docs/Web/API/SpeechSynthesis/onvoiceschanged
So, when we want to speak a piece of data, due to user interaction, and not on page load as it appears, we need to do that when the voiceschanged event is fired. If we speak without waiting for this event, then the first utterance will be spoken via the default TTS, and I am not sure why this happens in fact.
I think this still not so clear, even to me 😊, but this is the result I got after some experimenting with this. Please let  me know what do you think!

File: ts/app.ts (hunk 2 above)
Comment (mohammad-suliman) [reply ts/app.ts]: In fact, if I understand what you have said, yes. I want this code to be executed on page load. We want to first set the focus to the input field, and then populate the list of available TTSs when the page loads. So, is this not what we are getting this way?

File: ts/view/audio.ts (hunk 1 above)
Comment (mohammad-suliman) [reply ts/view/audio.ts]: Actually, you are rite. We have some announcement which happens on page load, not sure why. It seems that your intuition is correct. I can try to fix the issue of this announcement if the approach is acceptable from your side.

Comment (oferb) [reply ts/view/audio.ts]: I think you can just speak the default TTS, and when it loads, change the TTS.
It will probably load before the user interacts with the page.
But - if there's ever a problem and it never loads, at least we still use the default TTS (and not, say, stay silent).

Only problem maybe is that the TTS changes in the middle of the speech. I think it's not a problem, but, I say let's hope for the best and if we see it's a problem, we fix it then.

File: ts/app.ts (hunk 2 above)
Comment (oferb) [reply ts/app.ts]: It was already on page load of index.html.
Now it's on page load of every page that loads the script.
Which pages do you want it to run on page load on?

**Comment by oferb** [comment]
Approve
